### PR TITLE
feat(dynatrace): serialize metrics using dt metrics utils

### PR DESCRIPTION
**Pull request by dyladan**
This is the first part of splitting #4853 into multiple PRs

It introduces the dynatrace metrics utility library for metric serialization. This library handles normalization and serialization of metric names, attributes, and values in a manner consistent with other Dynatrace metric integrations.

The bulk of this PR is removing serialization/normalization code which is handled by the metric library.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@dyladan ignore, but you need to have a review from your codeowners :D

**Review by arminru — APPROVED** [review]
@bogdandrutu Sure! This was already reviewed extensively during the internal review of #4853 where this was split from.

File: exporter/dynatraceexporter/config/config.go
Comment (mx-psi) [inline exporter/dynatraceexporter/config/config.go:60]: More user friendly if you use the YAML name?
```suggestion
			return errors.New("api_token is required if Endpoint is provided")
```

**Review by mx-psi — APPROVED** [review]

**Pull request merged**
